### ISSUE TITLE
Borrow Checker False Positives

### ISSUE THREAD

**Issue by stonecodekiller** [intellij-rust/intellij-rust#10777]
## Environment

* **IntelliJ Rust plugin version:** 0.4.196.5423-231
* **Rust toolchain version:** 1.71.0-nightly (2a8221dbd 2023-05-11) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2023.1.5 (CL-231.9225.21)
* **Operating system:** Linux 5.15.114-1-manjaro
* **Macro expansion:** enabled

## Problem description

The borrow checker of the Rust plugin seems to be flagging any uses of a reference inside a function as "cannot move" despite being a primitive type that has Copy. 

## Steps to reproduce

```
struct S {
    x: i32,
}

fn show_issue(issue: &S) -> i32 {
    issue.x * 3
}
```

![borrow_checker_issue](https://github.com/intellij-rust/intellij-rust/assets/117553715/fe3d68ec-6f77-4c12-a60a-381c92420c24)

**Comment by Maksim20023** [comment]
Hi! Thank you for reaching us.
This issue does not reproduce in the current environment: CLion 2023.2, Rust plugin 0.4.199.5415-232 because it has been fixed: #10528 . Please, update your [IDE](https://www.jetbrains.com/help/clion/update.html) and [plugin](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-quick-start.html#update), and try to reproduce the issue.

**Comment by 0xcaff** [comment]
This still occurs for me in the latest version of rust rover, it is very annoying to read code and see false positives on almost every line.

Using version of plugin bundled 233.20527.212 and rust rover 2023.3 EAP. This has been broken in clion for as long as I could remember too.
